cdap-runtime-ext-remote-hadoop/src/main/java/io/cdap/cdap/runtime/spi/provisioner/remote/RemoteHadoopProvisioner.java

```diff
@@ -145,17 +147,29 @@ String selectEdgeNode(String hostConfigValue, String profileName, ProgramRunInfo
       if (LOG.isTraceEnabled()) {
         LOG.trace("All profile counters are: " + LATEST_EDGE_NODE.toString());
       }
-      host = hosts.get(counterValue % hosts.size());
+      selectedHost = selectCheckedEdgeNode(hosts, counterValue, conf.getEdgeNodeCheck(), conf.getTimeout());
```

Comment:
and why we renamed host to selectedHost?  the sense behind of the variable not really changed, only the way how the value come to it.